gamemode/modules/party/sv_party.lua

```diff
@@ -0,0 +1,322 @@
+RVR.Party = RVR.Party or {}
+local party = RVR.Party
+party.idCounter = party.idCounter or 0
+
+-- TODO: Start player as spectator
+
+party.JOIN_MODE_PUBLIC = 0
+party.JOIN_MODE_STEAM_FRIENDS = 1
+party.JOIN_MODE_INVITE_ONLY = 2
+
+party.joinModeStrs = {
+    [0] = "public",
+    [1] = "steam-friends only",
+    [2] = "invite only",
+}
+
+util.AddNetworkString( "RVR_Party_updateClient" )
+util.AddNetworkString( "RVR_Party_createParty" )
+util.AddNetworkString( "RVR_Party_updateFriends" )
+
+local function updateClientPartyData( id )
+    local partyData = party.getParty( id )
+
+    net.Start( "RVR_Party_updateClient" )
+        net.WriteUInt( id, 32 )
+        net.WriteBool( tobool( partyData ) )
+        if partyData then
+            net.WriteTable( partyData )
+        end
+    net.Broadcast()
+end
+
+function party.createParty( partyName, owner, tag, color, joinMode )
+    for id, partyData in pairs( party.parties ) do
+        if partyData.name == partyName then
+            return nil, "Party with name " .. partyName .. " already exists"
+        end
+        if partyData.tag == tag then
+            return nil, "Party with tag " .. tag .. " already exists"
+        end
+    end
+
+    if owner:GetPartyID() then
+        return nil, "Player " .. owner:Nick() .. " already in a party"
+    end
+
+    if #tag ~= 4 then
+        return nil, "Tag must be 4 characters"
+    end
+
+    if #partyName < 5 or tonumber( partyName ) then
```

Comment:
Pull out min and max party name length into config, check for whitespace only names and/or just trim it